### DISCUSSION TITLE
Helper Tool Stopping App from Working.

### DISCUSSION THREAD

**Discussion by RedJobson** [SelfControlApp/selfcontrol#877] [Q&A]
Hi Internet. I am running Self Control Version 4.0.2 (410) on Mac (21.5-inch, Mid 2011) MacOS Hugh Sierra 10.13.6 (17G14042).

When I try to start a block I get a window saying -

''Self Control is trying to install a new helper tool'' My options are 'Install Helper' or Cancel. 

If I click 'Install Helper' I get another window saying - 

'Couldn’t communicate with a helper application. Try your operation again. If that fails, quit and relaunch the application and try again.'

I am stumped. Any suggestions? Thank you in advance :-)

**Comment by cstigler** [comment]
Hi @RedJobson, unfortunately not sure why it'd be failing, but a few things you can try:

1. Restart your computer (I know this is basic advice, but... it works)
2. Make sure that you're logged in as an administrator account on your computer
3. Try deleting SelfControl and reinstalling it (then restarting again)

Hope you get it working!

**Reply by RedJobson** [reply]
Hi @cstigler Thanks for your reply.

I tried / checked all those and still no luck. It's determined to install this helper tool. Still getting the same issues as in my original post ... :-/

**Reply by puzzledoyster** [reply]
Just wanted to second that I have the exact problem too. 

Some potential clues: it was fine on my old MacBook (2015), but when I migrated the data to my new macbook it started causing difficulties. In addition: there were some issues transferring my data to the new macbook because certain startup items in my MacBook was apparently jamming up the login, and that had to be disabled first, before things would start up normally. 

I wonder @cstigler : does the so-called helper application have anything to do with what's been called "org.eyebeam.SelfControl"? Or is it something else? I'm going to look for hidden files and see if there are things related to this helper application that need deleting in order to properly reinstall the app. Because when I reinstalled it, it clearly remember my old blocklist.

**Reply by cstigler** [reply]
Hi @puzzledoyster, yes the helper tool SelfControl uses is `org.eyebeam.SelfControl`. In terms of deleting SelfControl settings, here are some commands that will delete them:

1. Close SelfControl
2. `sudo launchctl remove org.eyebeam.selfcontrold`
3. `sudo find -E /usr/local/etc -regex '.*\/\.[a-z0-9]{40}\.plist' -delete -print`
4. `find -E ~/Library/Preferences -regex '.*\/\.[a-z0-9]{40}\.plist' -delete -print`
5.  `defaults delete org.eyebeam.SelfControl`

Unfortunately can't help debug in more depth than that, but hope that helps!

**Comment by RedJobson** [comment]
Hey, how did you get on with the fix?

**Reply by puzzledoyster** [reply]
It succeeded in deleting some of the pre-saved settings on the computer, but still was not able to resolve the issue. So something odd is going on, beyond just pre-saved settings. 

This is quite strange, but here's a thought. Just googling "Couldn’t communicate with a helper application", this appears to be a general problem in MacBook systems. So one line of attack is to look through the various forums and see what worked for other people, experiment and see if that works for us with this particular app.

**Reply by h0h0h0** [reply]
> So one line of attack is to look through the various forums and see what worked for other people, experiment and see if that works for us with this particular app.

going through this issue now; also getting that helper request popup.
